stake/src/status.rs

```diff
@@ -27,34 +27,48 @@ impl From<ValidatorStatus> for pb::ValidatorStatus {
         pb::ValidatorStatus {
             identity_key: Some(v.identity_key.into()),
             voting_power: v.voting_power,
-            state: match v.state {
-                ValidatorState::Inactive => pb::validator_status::ValidatorState::Inactive,
-                ValidatorState::Active => pb::validator_status::ValidatorState::Active,
-                ValidatorState::Unbonding { .. } => pb::validator_status::ValidatorState::Unbonding,
-                ValidatorState::Slashed => pb::validator_status::ValidatorState::Slashed,
-            } as i32,
-            unbonding_epoch: match v.state {
-                ValidatorState::Unbonding { unbonding_epoch } => Some(unbonding_epoch),
-                _ => None,
-            },
+            // TODO: Why is this becoming an Option type? the protobuf type
+            // doesn't define it as optional, and it should always be present.
```

Comment:
This is becoming a Rust option because all proto fields are optional; primitive types have default values if missing, but proto messages are either there or not, so the Rust types get filled up with `Option`s.  I think the `optional` field in proto3 just records extra info that lets the application determine whether a primitive type was really missing or whether it was just a default value.